src/components/Tabs/Tab.js

```diff
@@ -0,0 +1,28 @@
+import React, { useState } from 'react';
+
+import { StyledTab } from './Tab.styled';
+
+const Tab = (props) => {
+  const [activeTab, setActiveTab] = useState(false);
+
+  const changeActiveStatus = (e) => {
+    console.log(typeof props.active);
+
+    // props.active === false;
+    setActiveTab();
+    setActiveTab(!activeTab);
+  };
+
+  return (
+    <StyledTab
+      href={props.href}
+      title={props.title}
+      active={activeTab}
+      onClick={changeActiveStatus}
+    >
+      {props.children}
```

Comment:
Jeśli struktura będzie wyglądać jak sugerowałęm wcześniej to w props.children będziesz miał wszystkie dane.
Wystarczy je wylistować i podpiąć "klikanie" w zakładki. Przechowywanie aktywnej zrobisz tak jak teraz i wszystko powinno działać jak należy.

Tutaj więcej szczegółów znajdziesz: 
https://www.smashingmagazine.com/2021/08/react-children-iteration-methods/

Jak za mało to szukać pod: `Children.map` lub `Children.forEach`